src/alpaka/AlpakaCore/AtomicPairCounter.h

```diff
@@ -0,0 +1,53 @@
+#ifndef HeterogeneousCore_CUDAUtilities_interface_AtomicPairCounter_h
+#define HeterogeneousCore_CUDAUtilities_interface_AtomicPairCounter_h
+
+#include <cstdint>
+
+namespace cms {
+  namespace Alpaka {
```

Comment:
To be compatible with CMSSW convention
```suggestion
  namespace alpaka {
```

Reply:
Ha yes, the issue is that when I rename cms::Alpaka to cms::alpaka, any function from alpaka library (for example here, alpaka::atomic::atomicOp) is looked for within cms::alpaka.
Hmm maybe not ideal to have the same name here?

For Kokkos, an issue with cms::kokkos does not appear, because the namespace from Kokkos has an upperletter: Kokkos::atomic_fetch_add

Reply:
> Ha yes, the issue is that when I rename cms::Alpaka to cms::alpaka, any function from alpaka library (for example here, alpaka::atomic::atomicOp) is looked for within cms::alpaka.
> Hmm maybe not ideal to have the same name here?

Fair point. I'd expect qualifying the Alpaka function calls with leading `::` along `::alpaka::atomic::atomic::atomicOp` to work though.

Reply:
Hmm yes there must be a way around, but then whenever we are within cms::alpaka and we call a function from alpaka library, we need to use it, which makes it not very elegant, no?

Reply:
I agree having to use leading `::` is slightly annoying, but so would be to remember that one of our namespaces starts with capital letter, and, conversely, that the name `cms::Alpaka` is a namespace and not a class. Thinking about a possible use in CMSSW, I'd expect `cms::[aA]lpaka` to be used more than `::alpaka` inside functions/classes in `cms::[aA]lpaka`, and therefore I'd value interface consistency over slight annoyance in utility function/class implementations.

Written that, this project is all about prototyping and certain shortcuts are allowed, and the choice of the namespace name is not that relevant for the question "what would it take to use Alpaka in CMSSW" (even if it is part of it). I could accept the capital version now, and it would be straightforward to change it later, but it would be easier to do it now than later.

Reply:
@makortel Ok thanks for the explanation, yes will change it to `cms::alpaka` and just use `::alpaka` for alpaka library calls then.

Reply:
Ok I have just pushed sth at https://github.com/cms-patatrack/pixeltrack-standalone/pull/160/commits/8e026919b02f298c1392c5d6396e1acd330a2aef to get rid of the cms::Alpaka namespace, and follow CMSSW conventions.
In the end, I found it nice to simply rename `cms::Alpaka` to describe better what was behind the "Alpaka" anyway: `cms::Alpaka` namespace was only used in `AlpaCore` for helper functions, hence I have renamed `cms::Alpaka` to `cms::alpaka_helpers`.
This is in the same spirit as other 'home-made' alpaka namespaces we already have for other purposes: `alpaka_common`, `alpaka_cuda_async`, etc.
Let me know if you are ok with this solution, or I should rather change the alpaka library namespace.

Reply:
The namespaces in `AlpakaCore/alpakaConfig.h` special and should not be used outside explicitly (with the exception of `alpaka_common` that appears to be useful in #160). The macro `ALPAKA_ACCELERATOR_NAMESPACE` should be used instead, and it gets replaced with the namespace corresponding the backend for which the translation unit is being compiled for.

That reminds me (sorry for not before) that for `sycltest` program we had a similar discussion on the namespaces in #102 (ok, I see the discussion is not in the PR but I'm pretty sure I discussed about that with @fwyzard; found it in our private chat). There we decided to go with `cms::sycltools` and figure out better name later (if needed). So how about `cms::alpakatools`? I could accept `alpaka_helpers" as well for now, but I wanted ask about `cms::alpakatools` first.

Reply:
Ok sure, will put cms::alpakatools rather than cms::alpaka_helpers if you prefer, will make it more consistent with the sycl side

Reply:
@makortel Just renamed from cms::alpaka_helpers to cms::alpakatools as you suggested (+ few other cleanings, templating) (I put the changes in https://github.com/cms-patatrack/pixeltrack-standalone/pull/160 so that all the code is directly properly modified).